pkg/reconciler/tenancy/workspacemounts/workspacemounts_reconcile.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2024 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package workspacemounts
+
+import (
+	"context"
+	"strings"
+
+	"github.com/kcp-dev/logicalcluster/v3"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	utilserrors "k8s.io/apimachinery/pkg/util/errors"
+
+	tenancyv1alpha1 "github.com/kcp-dev/kcp/sdk/apis/tenancy/v1alpha1"
+)
+
+type reconcileStatus int
+
+const (
+	reconcileStatusStopAndRequeue reconcileStatus = iota
+	reconcileStatusContinue
+)
+
+type reconciler interface {
+	reconcile(ctx context.Context, workspace *tenancyv1alpha1.Workspace) (reconcileStatus, error)
+}
+
+// reconcile reconciles the workspace objects. It is intended to be single reconciler for all the
+// workspace replated operations. For now it has single reconciler that updates the status of the
+// workspace based on the mount status.
+func (c *Controller) reconcile(ctx context.Context, ws *tenancyv1alpha1.Workspace) (bool, error) {
+	getMountObjectFunc := func(ctx context.Context, cluster logicalcluster.Path, ref *v1.ObjectReference) (*unstructured.Unstructured, error) {
+		resourceName := strings.ToLower(ref.Kind) + "s"
```

Comment:
should we maybe reference resources? This logic here will break if somebody has custom plural.

Reply:
I was trying to avoid hardcoding and make a generic one.
Alternative is either hardcode in the core, which I tried to avoid so code does not need to be changed when new mount implementation is done.
Not sure if there is another easy'ish alternative?

Reply:
Idea was if you use Plural already - that's kinda wrong naming

Reply:
true. This really needs a rest mapper, but a per-workspace rest mapper is also ugly 🤔

Reply:
I think this gives us a path forward, for now, we can change this later while we iterate.